src/codecs/avif/encoder.rs

```diff
@@ -67,24 +66,21 @@ impl<W: Write> AvifEncoder<W> {
         let quality = min(quality, 100);
         let speed = min(speed, 10);
 
+        let encoder = Encoder::new()
+            .with_quality(f32::from(quality))
+            .with_alpha_quality(f32::from(quality))
+            .with_speed(speed);
+
         AvifEncoder {
             inner: w,
-            fallback: vec![],
-            config: Config {
-                quality: f32::from(quality),
-                alpha_quality: f32::from(quality),
-                speed,
-                premultiplied_alpha: false,
-                color_space: ravif::ColorSpace::RGB,
-                // match core count
-                threads: 0,
-            },
+            encoder,
         }
     }
 
     /// Encode with the specified `color_space`.
     pub fn with_colorspace(mut self, color_space: ColorSpace) -> Self {
-        self.config.color_space = color_space.to_ravif();
+        let encoder = self.encoder;
+        self.encoder = encoder.with_internal_color_space(color_space.to_ravif());
```

Comment:
Does the borrow checker allow you to write `self.encoder = self.encoder.with_internal_color_space(...)`? If so, that would be cleaner.

Reply:
It does.